### PR TITLE
Sign up form

### LINKED ISSUE #12: Start Setting up the Sign Up Form
Users will need need to fill out a couple text fields to sign up for an event. For this task, you will set up the form that should be necessary to sign up. This task was inspired by the existing sign up page [here](https://www.cityfarmslo.org/volunteer), but please don't worry about any styling yet!

**Requirements**

- [x] Using the text fields from the [MUI library](https://material.io/components/text-fields), set up a form that accepts a name, email, phone number, and any notes. Please note you may have to install the mui library using npm if it hasn't already been installed.
- [x] For now, console.log all of the info in a single JS object when the user clicks the submit button

**Extra Credit**

- [ ] Using an NPM package like react-hook-form or formik, add validation and small error messages if the user doesn't fill out a required field or formats their email or phone number wrong

### PR DESCRIPTION
**Pull request by adellevo**
* added initial form setup, still need to do validation

### CODE REVIEW

**Comment by adellevo** [comment]
Notes:

- Prints in console when all fields are properly filled out

Screen Shot 2022-01-11 at 11 40 30 PM

- Some example validation: number/name/email are required fields, number/email need to follow a specific format

Screen Shot 2022-01-11 at 11 38 10 PM
Screen Shot 2022-01-11 at 11 39 25 PM

Unresolved Issues:

- Somehow keep getting this error 
Screen Shot 2022-01-11 at 11 42 22 PM

- Found multiple articles referencing this issue with the same suggestion for how to fix it (this is one of them: https://stackoverflow.com/questions/57558219/dealing-with-failed-to-load-plugin-react-declared-in-eslintrc-cannot-find) but changing my settings.json file didn't seem to work. Since I kept getting the compilation error, I had to x out of the error every time I reloaded the browser to test my code. 

Screen Shot 2022-01-11 at 11 47 34 PM

File: frontend/src/components/SignUpForm.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    const validationSchema = Yup.object().shape({
+        name: Yup.string().required("Required field"),
+        email: Yup.string()
+            .required("Required field")
+            .email("Invalid format"),
+        number: Yup.string()
+            .required("Required field")
+            .min(10, "Too short")
+            .max(10, "Too long")
+            .matches(phoneRegExp, "Invalid format")
```
Comment (vwinstea) [inline frontend/src/components/SignUpForm.jsx:29]: The validation for phone numbers works great, but I feel like the error messages could be more helpful. I was originally putting dashes in my phone number (111-222-3333) and was initially confused about why my phone number was too long.

**Review by vwinstea — APPROVED** [review]
Great work! The form input works great and the validation works fantastically! The only small note I have for next time is that for some reason your material ui dependencies were added to package.json but yup and @hookform/resolvers weren't for some reason, so just check your dependencies before you push. Other than that, this is all really good stuff!

**Comment by vwinstea** [comment]
That ESLint bug is weird, I just checked our dependencies and it looks like `eslint-plugin-react` is in there, so idk why your machine can't find it. I'll let you know if anyone else on our team runs into the same issue

**Pull request merged**